source/vibe/web/rest.d

```diff
@@ -618,6 +660,88 @@ private HTTPServerRequestDelegate jsonMethodHandler(alias Func, size_t ridx, T)(
 	return &handler;
 }
 
+/**
+ * Generate an handler that will wrap the server's method
+ *
+ * This function returns an handler that handles the http OPTIONS method.
+ *
+ * It will return the ALLOW header with all the methods on this resource
+ * And it will handle Preflight CORS.
+ *
+ * Params:
+ *	routes = a range of Routes were each route has the same resource/URI
+ *				just different method.
+ *	settings = REST server configuration.
+ *
+ * Returns:
+ *	A delegate suitable to use as an handler for an HTTP request.
+ */
+private HTTPServerRequestDelegate optionsMethodHandler(RouteRange)(RouteRange routes, RestInterfaceSettings settings = null)
+{
+	import vibe.http.server : HTTPServerRequest, HTTPServerResponse;
+	import std.algorithm : map, joiner, any;
+	import std.conv : text;
+	import std.array : array;
+	import vibe.http.common : httpMethodString, httpMethodFromString;
+	// NOTE: don't know what is better, to keep this in memory, or generate on each request
```

Comment:
Should be good as it is now. In case of such small allocations, I'd always opt to reduce the per-request GC pressure (even if there remains a lot to do for the REST module in that regard).